onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -0,0 +1,695 @@
+        ID3D12Resource* inputResource,
+        ID3D12Resource* outputResource,
+        DFTShaderConstants& constants,
+        ID3D12GraphicsCommandList* commandList)
+    {
+        D3D12_RESOURCE_BARRIER uav_barriers[2];
+        uav_barriers[0] = CD3DX12_RESOURCE_BARRIER::UAV(inputResource);
+        uav_barriers[1] = CD3DX12_RESOURCE_BARRIER::UAV(outputResource);
+        commandList->ResourceBarrier(2, uav_barriers);
+        // Set resource views
+        commandList->SetComputeRootUnorderedAccessView(
+            0, // root parameter index
+            inputResource->GetGPUVirtualAddress()
+        );
+
+        commandList->SetComputeRootUnorderedAccessView(
+            1, // root parameter index
+            outputResource->GetGPUVirtualAddress()
+        );
+        auto pendingElementCount = constants.ElementCount;
+
+        // Dispatch up to the maximum number of threads per iteration until
+        // all elements are completed
+        while (pendingElementCount > 0)
+        {
+            constants.StartIndex = constants.ElementCount - pendingElementCount;
+
+            uint32_t dispatchSizeX;
+
+            DFTHelpers::GetNextDispatchSize(
+                pendingElementCount,
+                1,
+                64,
+                dispatchSizeX,
+                pendingElementCount
+            );
+
+            // Set root constants
+            commandList->SetComputeRoot32BitConstants(
+                2, // root parameter index
+                21, // Constant count
+                &constants,
+                0 // offset
+            );
+
+            commandList->Dispatch(dispatchSizeX, 1, 1);
+        }
+
+        commandList->ResourceBarrier(2, uav_barriers);
+    }
+
+    void BluesteinZChirp(
+        ID3D12Resource* /*inputResource*/,
+        ID3D12Resource* /*outputResource*/,
+        ID3D12GraphicsCommandList* /*commandList*/)
+    {
+        ORT_THROW_HR(E_NOTIMPL);
+    }
+};
+
+struct DFTShapeInferrer : public WRL::Base<IMLOperatorShapeInferrer>
+{
+    STDMETHOD(InferOutputShapes)(IMLOperatorShapeInferenceContext* context) noexcept
+    {
+        try
+        {
+            int64_t axis;
+            ORT_THROW_IF_FAILED(context->GetAttribute("axis", MLOperatorAttributeType::Int, 1, sizeof(int64_t), reinterpret_cast<void*>(&axis)));
+            int64_t isInverseInt;
+            ORT_THROW_IF_FAILED(context->GetAttribute("inverse", MLOperatorAttributeType::Int, 1, sizeof(int64_t), reinterpret_cast<void*>(&isInverseInt)));
+            int64_t isOnesidedInt;
+            ORT_THROW_IF_FAILED(context->GetAttribute("onesided", MLOperatorAttributeType::Int, 1, sizeof(int64_t), reinterpret_cast<void*>(&isOnesidedInt)));
+            bool isOnesided = static_cast<bool>(isOnesidedInt);
+            bool inverse = static_cast<bool>(isInverseInt);
+
+            if (inverse && isOnesided) {
+                throw new std::exception("isOnesided and inverse attributes cannot be enabled at the same time");
```

Comment:
>isOnesided [](http://example.com/codeflow?start=42&length=10)

'onesided' is the actual ONNX attribute name for the reader. isOnesided is just the internal local variable.